Parser/pegen.c

```diff
@@ -400,16 +397,17 @@ _PyPegen_raise_error_known_location(Parser *p, PyObject *errtype,
 
     if (!error_line) {
         Py_ssize_t size = p->tok->inp - p->tok->buf;
-        if (size && p->tok->buf[size-1] == '\n') {
-            size--;
-        }
         error_line = PyUnicode_DecodeUTF8(p->tok->buf, size, "replace");
         if (!error_line) {
             goto error;
         }
     }
 
-    Py_ssize_t col_number = byte_offset_to_character_offset(error_line, col_offset);
+    Py_ssize_t col_number = col_offset;
+
+    if (p->tok->encoding != NULL) {
```

Comment:
Question: I had something similar to this, but with `p->tok->enc` instead, which wasn't really working. What's the difference between the two?

Reply:
`p->tok->enc` is the encoding for the current string, while `encoding` is the encoding for the entire source.

Reply:
Oh, okay. Turns out I should have just read the comments in `tokenizer.h`. Thanks!